function/func_error.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package function
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+// NewFuncError returns a new function error with the
+// given message.
+func NewFuncError(text string) *FuncError {
+	return &FuncError{
+		Text: text,
+	}
+}
+
+// NewArgumentFuncError returns a new function error with the
+// given message and function argument.
+func NewArgumentFuncError(functionArgument int64, text string) *FuncError {
+	return &FuncError{
+		Text:             text,
+		FunctionArgument: &functionArgument,
+	}
+}
+
+// FuncError is an error type specifically for function errors.
+type FuncError struct {
+	Text             string
```

Comment:
Apologies, I had commented about the variable name of `text` while forgetting the underlying struct field was named `Text` 😆

Reply:
No worries. Have left it as `errMsg`.

Reply:
Nit: I know its trivial in this case, but can we please ensure all exported types and fields have Go documentation? For example, the website documentation for these fields seems great.